### PR TITLE
[Fix #7513] Fix abrupt error on autocorrecting with `--disable-uncorrectable`

### LINKED ISSUE #7513: undefined method `location' for nil:NilClass @ cop.rb:176
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior

Describe here how you expected RuboCop to behave in this particular situation

To do the `"rubocop --auto-correct --disable-uncorrectable"` command and finish the analysis and correction properly

## Actual behavior
Describe here what actually happened.

An error is prompted

`106 files inspected, 9554 offenses detected, 9526 offenses corrected
undefined method `location' for nil:NilClass
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cop/cop.rb:176:in `disable_uncorrectable'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cop/cop.rb:161:in `correct'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cop/cop.rb:134:in `add_offense'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cop/lint/syntax.rb:35:in `add_offense_from_diagnostic'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cop/lint/syntax.rb:24:in `block in offenses_from_processed_source'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cop/lint/syntax.rb:23:in `each'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cop/lint/syntax.rb:23:in `offenses_from_processed_source'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cop/team.rb:39:in `inspect_file'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:290:in `inspect_file'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:238:in `block in do_inspection_loop'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:270:in `block in iterate_until_no_changes'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:263:in `loop'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:263:in `iterate_until_no_changes'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:234:in `do_inspection_loop'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:119:in `block in file_offenses'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:137:in `file_offense_cache'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:117:in `file_offenses'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:108:in `process_file'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:87:in `block in each_inspected_file'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:86:in `each'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:86:in `reduce'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:86:in `each_inspected_file'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:73:in `inspect_files'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/runner.rb:39:in `run'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cli.rb:213:in `execute_runner'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cli.rb:80:in `execute_runners'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/lib/rubocop/cli.rb:51:in `run'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/exe/rubocop:13:in `block in <top (required)>'
/home/djigo/.rvm/rubies/ruby-2.5.5/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/home/djigo/.rvm/gems/ruby-2.5.5/gems/rubocop-0.76.0/exe/rubocop:12:in `<top (required)>'
/home/djigo/.rvm/gems/ruby-2.5.5/bin/rubocop:23:in `load'
/home/djigo/.rvm/gems/ruby-2.5.5/bin/rubocop:23:in `<main>'
/home/djigo/.rvm/gems/ruby-2.5.5/bin/ruby_executable_hooks:24:in `eval'
/home/djigo/.rvm/gems/ruby-2.5.5/bin/ruby_executable_hooks:24:in `<main>'`

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

Run `"rubocop --auto-correct --disable-uncorrectable" 
`
## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:
0.76.0 (using Parser 2.6.5.0, running on ruby 2.5.5 x86_64-linux)

```
$ [bundle exec] rubocop -V
0.76.0 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)
```

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes #7513

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/cop.rb [hunk 1]
```diff
@@ -147,8 +147,7 @@ def duplicate_location?(location)
       end
 
       def correct(node)
-        reason = reason_to_not_correct(node)
-        return reason if reason
+        return if reason_to_not_correct(node)
```
Comment (tejasbubane) [inline lib/rubocop/cop/cop.rb]: This is for preserving class line-length of `182`: https://github.com/rubocop-hq/rubocop/blob/e97eb5ce74b3319c21de05f6c1fd78493bd55b47/.rubocop_todo.yml#L15-L16

Comment (buehmann) [reply lib/rubocop/cop/cop.rb]: This changes the return value and the status of the offense in turn. It is strange that there is no test coverage for this.

Comment (tejasbubane) [reply lib/rubocop/cop/cop.rb]: Ha! my bad.

**Comment by koic** [comment]
This looks good to me. Could you rebase with the latest master?

**Comment by tejasbubane** [comment]
@koic Rebased.

**Comment by koic** [comment]
Thanks!

**Pull request merged**